PieceOfShift/Chart.qml

```diff
@@ -0,0 +1,151 @@
+                parent.forceActiveFocus()
+            }
+            onExited: {
+                hoverEnabled = true
+            }
+            onWheel: {
+                if((wheel.angleDelta).y > 0){
+                   var p = chartview.plotArea
+                   var zoom = 4/3
+                    chartview.zoomIn(Qt.rect(p.x,p.y + (p.height - p.height/zoom),p.width / zoom, p.height / zoom))
+                }
+                else{
+                    p = chartview.plotArea
+                    zoom = 0.75
+                    chartview.zoomIn(Qt.rect(p.x ,p.y + (p.height - p.height/zoom),p.width / zoom, p.height / zoom))
+                }
+            }
+            onPressed: {
+                timer.mouse_x_= mouseX
+                timer.mouse_y_= mouseY
+                timer.start()
+            }
+            onReleased: {
+                timer.stop()
+
+            }
+            Timer{
+                id: timer
+                repeat: true
+                interval: 50
+                property int mouse_x_: chartMouseArea.mouseX
+                property int mouse_y_: chartMouseArea.mouseY
+                onTriggered: {
+                    chartview.scrollUp(chartMouseArea.mouseY - mouse_y_)
+                    chartview.scrollLeft(chartMouseArea.mouseX - mouse_x_)
+                    mouse_x_ = chartMouseArea.mouseX
+                    mouse_y_ = chartMouseArea.mouseY
+                }
+            }
+
+        }
+
+
+        LineSeries{
+            id: lineseries
+            axisX: x_axis
+            axisY: y_axis
+            pointsVisible: true
+
+            ValueAxis{
+                id: y_axis
+                labelsColor: "white"
+                color: "white"
+                min: 0
+                max: 10
+                titleText: ""
+            }
+            ValueAxis{
+                id: x_axis
+                labelsColor: "white"
+                color: "white"
+                min: 0
+                max: 10
+                titleText: ""
+            }
+            XYPoint { x: 0; y: 0 }
+            XYPoint { x: 1.1; y: 2.1 }
+            XYPoint { x: 1.9; y: 3.3 }
+            XYPoint { x: 2.1; y: 2.1 }
+            XYPoint { x: 2.9; y: 4.9 }
+            XYPoint { x: 3.4; y: 3.0 }
+            XYPoint { x: 4.1; y: 3.3 }
+            onHovered: {
+                //
+                var punkt = "(" + point.x.toFixed(2) + "," + point.y.toFixed(2) + ")"
+                _text.text = punkt
+                var p = chartview.mapToPosition(Qt.point(point.x,point.y),lineseries)
+                rectangle1.visible = true
+                rectangle1.x = p.x
+                rectangle1.y = p.y
+            }
+
+            onPointAdded: {
+                var new_point = at(index)
+                if(new_point.x > x_axis.max){
+                    x_axis.max*=1.5
+                }
+                if(new_point.y > y_axis.max){
+                    y_axis.max*=1.5
+                }
```

Comment:
Would it be better to set x/y_axis.max = newpoint.x/y the new point always is in view?